controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -589,17 +590,49 @@ func (w *Workload) UpdateKubeProxyImageInfo(ctx context.Context, kcp *controlpla
 	if len(ds.Spec.Template.Spec.Containers) == 0 {
 		return nil
 	}
-	newImageName, err := util.ModifyImageTag(ds.Spec.Template.Spec.Containers[0].Image, kcp.Spec.Version)
+
+	parsedImage, err := reference.ParseNormalizedNamed(ds.Spec.Template.Spec.Containers[0].Image)
```

Comment:
Potential issue for followup: this should likely iterate over the containers like the coreDNS implementation is doing.